plugins/Jobs.cpp

```diff
@@ -220,128 +238,192 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
+                          << job["name"] << ", removing attribute.");
+                    job.erase("repeat");
+                } else if (!_validateRepeat(job["repeat"])) {
+                    throw "402 Malformed repeat";
+                }
             }
-        }
 
-        // We'd initially intended for any value to be allowable here, but for
-        // performance reasons, we currently will only allow specific values to
-        // try and keep queries fast. If you pass an invalid value, we'll throw
-        // here so that the caller can know that he did something wrong rather
-        // than having his job sit unprocessed in the queue forever. Hopefully
-        // we can remove this restriction in the future.
-        if (priority != 0 && priority != 500 && priority != 1000) {
-            throw "402 Invalid priority value";
-        }
-
-        // Are we creating a new job, or updating an existing job?
-        if (updateJobID) {
-            // Update the existing job.
-            if(!db.write("UPDATE JOBS SET "
-                           "repeat   = " + SQ(SToUpper(request["repeat"])) + ", " +
-                           "data     = JSON_PATCH(data, " + safeData + "), " +
-                           "priority = " + SQ(priority) + " " +
-                         "WHERE jobID = " + SQ(updateJobID) + ";"))
-            {
-                throw "502 update query failed";
-            }
-            content["jobID"] = SToStr(updateJobID);
-        } else {
-            // Normal jobs start out in the QUEUED state, meaning they are ready to run immediately.
-            // Child jobs normally start out in the PAUSED state, and are switched to QUEUED when the parent
-            // finishes itself (and itself becomes PAUSED).  However, if the parent is already PAUSED when
-            // the child is created (indicating a child is creating a sibling) then the new child starts
-            // in the QUEUED state.
-            auto initialState = "QUEUED";
+            // If no priority set, set it
+            int64_t priority = SContains(job, "priority") ? SToInt(job["priority"]) : JOBS_DEFAULT_PRIORITY;
+
+            // Validate that the parentJobID exists and is in the right state if one was passed.
+            int64_t parentJobID = SContains(job, "parentJobID") ? SToInt(job["parentJobID"]) : 0;
             if (parentJobID) {
-                auto parentState = db.read("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
-                if (SIEquals(parentState, "RUNNING")) {
-                    initialState = "PAUSED";
+                SQResult result;
+                if (!db.read("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";", result)) {
+                    throw "502 Select failed";
+                }
+                if (result.empty()) {
+                    throw "404 parentJobID does not exist";
```

Comment:
Sorry I forgot to mention this before, but perhaps we should alert here instead of throwing since we have these checks in peekCommand and so we should never reach this point.

I added similar checks in the retryAfter [PR](https://github.com/Expensify/Bedrock/pull/169/files#r116340952) and opted to log an alert instead of throwing in processCommand.

What do you think?